drive_webhook_receiver.py

```diff
@@ -0,0 +1,132 @@
+import os
+import boto3
+import traceback
+from googleapiclient.discovery import build
+from google.oauth2 import service_account
+from utils import fetch_ssm_params, _export_file
+from builder_client import BuilderClient
+from constants import PDF_PATH, SCOPES
+
+
+def handle_google_drive_changes(event, context):
+    """
+    A lambda that checks recent changes in google drive and updates related files in Builder.
+    """
+    try:
+        # Fetch the Builder API key, the guide and custom list IDs that we want to update, 
+        # the service account credentials, and the name of the folder in Google Drive
+        api_key, guide_and_list_ids, drive_folder_name, start_page, service_account_credentials = fetch_ssm_params()
+
+        builder_client = BuilderClient(api_key)
+
+        # Connect to google drive with credentials
+        creds = service_account.Credentials.from_service_account_info(service_account_credentials, scopes=SCOPES)
+        service = build('drive', 'v3', credentials=creds)
+
+        # Get the ID of the folder we are watching in google drive
+        folder_id_search = service.files().list(q=f"mimeType='application/vnd.google-apps.folder' and name='{drive_folder_name}'", pageSize=1, fields="nextPageToken, files(id, name)").execute()
+        folder_id = folder_id_search.get('files', [])[0].get('id')
+
+        # Check for recent changes
+        changes_response = service.changes().list(pageToken=start_page).execute()
+        changes = changes_response['changes']
+        while changes_response.get('nextPageToken', None):
+            changes_response = service.changes().list(pageToken=changes_response['nextPageToken']).execute()
+            changes.extend(changes_response['changes'])
+
+        for guide_id, customlist_id in guide_and_list_ids:
+            for change in changes:
+                # Get the file associated to the change in Google Drive
+                changed_file_id = change['fileId'] if change.get('fileId', None) else change['id']
+                changed_file = service.files().get(fileId=changed_file_id, fields='name, trashed, parents').execute()
+
+                # We can skip a change if it is not in the folder we are watching
+                if folder_id not in changed_file['parents']:
+                    continue
+
+                # Fetch the existing CustomListItem from Builder if it exists by filtering on the import_id field.
+                # This is needed to obtain the CustomListItem.id, which is required in the PATCH request
+                url = f"https://beta.guidebook.com/open-api/v1/custom-list-items?guide={guide_id}&custom_lists={customlist_id}&import_id={changed_file_id}"
+                response = builder_client.get(url)
+                custom_list_item = response.json()['results'][0] if len(response.json()['results']) > 0 else None
+
+                # If there is no existing custom list item, create a new one
+                if not custom_list_item:
```

Comment:
@cgoetze11 for readability purposes, it might be nice to break up each of the conditions to a separate function to keep the higher level logic a little more readable